plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaMetadata.java

```diff
@@ -282,7 +286,9 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
                 .filter(columnHandle -> !columnHandle.isInternal() && !columnHandle.isHidden())
                 .collect(toImmutableList());
 
-        checkArgument(columns.equals(actualColumns), "Unexpected columns!\nexpected: %s\ngot: %s", actualColumns, columns);
+        if (!supportsMissingColumnsOnInsert(tableHandle)) {
+            checkArgument(columns.equals(actualColumns), "Unexpected columns!\nexpected: %s\ngot: %s", actualColumns, columns);
```

Comment:
I don't think it is still `checkArgument`, maybe `verify` because we assume that engine will not call with different columns when `!supportsMissingColumnsOnInsert(tableHandle)`